### PR TITLE
implement user search

### PR DESCRIPTION
**Pull request by pbochis**

### CODE REVIEW

File: platform-server-rest/src/main/java/uno/cod/platform/server/rest/controller/UserController.java [hunk 1]
```diff
@@ -51,6 +51,12 @@ public UserController(UserService userService) {
         return new ResponseEntity<>(userService.update(dto, user), HttpStatus.OK);
     }
 
+    @RequestMapping(value = RestUrls.USER_SEARCH, method = RequestMethod.GET)
+    @PreAuthorize("isAuthenticated()")
+    public ResponseEntity<List<UserShowDto>> searchUsers(@RequestParam String searchValue){
+        return new ResponseEntity<>(userService.listUsersLikeNameOrEmail(searchValue), HttpStatus.OK);
```
Comment (Exordian) [inline platform-server-rest/src/main/java/uno/cod/platform/server/rest/controller/UserController.java]: this is a huge privacy issue

Comment (pbochis) [inline platform-server-rest/src/main/java/uno/cod/platform/server/rest/controller/UserController.java]: Do you mean this endpoint should only be accessible to companies?

**Comment by flowlo** [comment]
As @Exordian already mentioned, this raises privacy concerns. `UserShowDto` contains a lot of sensitive data (including e-mail address and full name). We'll not be exposing this. However we could implement a search that given name and/or e-mail address returns a username and/or ID of the user. This way we can allow users to search by name and/or e-mail address but will not reveal any sensitive information. WDYT?

**Comment by pbochis** [comment]
I think we need to expose both the full name and the email to people who want to search. I don't see why exposing the email address and the full name to a company representative who is searching for them would be a privacy concern.

**Comment by flowlo** [comment]
We should make that opt-in. E.g. the user should have to tick a flag to expose e-mail address and name to companies on Coduno.

**Comment by Exordian** [comment]
Until some point I thought we only expose the UserShowDto only to companies. Those would already know the Name etc from the application. If they register themselfs and/or can be searched by others we have to rethink that (e.g. with opt-in options). This was just the first commit I saw where one could really search our whole database with only an user account, which is clearly not what we want.

**Comment by flowlo** [comment]
Well, as a company I could also search for arbitrary (but likely) names and get a list of users that match in order to obtain personal data of people that I previously *did not* know. that's a an attack vector that needs either an opt-in from a users perspective, or we need to establish another relation between companies and users in order to really know which company knows about the existence of which user.

**Pull request closed without merging**